pytorch_lightning/trainer/evaluation_loop.py

```diff
@@ -291,24 +291,22 @@ def evaluate(self, model, dataloaders, max_batches, test_mode: bool = False):
             outputs = outputs[0]
 
         # give model a chance to do something with the outputs (and method defined)
-        model = self.get_model()
-
-        if test_mode and self.is_overriden('test_epoch_end'):
-            eval_results = model.test_epoch_end(outputs)
-        elif self.is_overriden('validation_epoch_end'):
-            eval_results = model.validation_epoch_end(outputs)
+        model = self.get_model()  # if you call evaluate on given model, ths getter gives None
```

Comment:
This gets all the tests passing for me:

```suggestion
        if isinstance(model, (LightningDistributedDataParallel, LightningDataParallel)):
            model = model.module
```

(you also need to add the appropriate imports)